src/python/pants/backend/python/lint/yapf/subsystem.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os.path
+from typing import Iterable, cast
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.option.custom_types import file_option, shell_str
+
+
+class Yapf(PythonToolBase):
+    options_scope = "yapf"
+    help = "A formatter for Python files (https://github.com/google/yapf)."
+
+    default_version = "yapf==0.31.0"
+    default_extra_requirements = ["setuptools", "toml"]
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.6"]
+    default_main = ConsoleScript("yapf")
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--skip",
+            type=bool,
+            default=False,
+            help=(
+                f"Don't use yapf when running `{register.bootstrap.pants_bin_name} fmt` and "
+                f"`{register.bootstrap.pants_bin_name} lint`."
+            ),
+        )
+        register(
+            "--args",
+            type=list,
+            member_type=shell_str,
+            help=(
+                "Arguments to pass directly to yapf, e.g. "
+                f'`--{cls.options_scope}-args="--no-local-style"`. '
+                "Certain arguments, specifically `--recursive`, `--in-place`, and "
+                "`--parallel`, will be ignored because Pants takes care of finding "
+                "all the relevant files and running the formatting in parallel."
+            ),
+        )
+        register(
+            "--config",
+            type=file_option,
+            default=None,
+            advanced=True,
+            help=(
+                "Path to style file understood by yapf "
+                "(https://github.com/google/yapf#formatting-style/).\n\n"
+                f"Setting this option will disable `[{cls.options_scope}].config_discovery`. Use "
+                f"this option if the config is located in a non-standard location."
+            ),
+        )
+        register(
+            "--config-discovery",
+            type=bool,
+            default=True,
+            advanced=True,
+            help=(
+                "If true, Pants will include any relevant config files during "
+                "runs (`.style.yapf`, `pyproject.toml`, `setup.cfg`, and `~/.config/yapf/style`)."
```

Comment:
We do not want to load global config files, it increases the risk of Works On My Machine errors. Pants can't do anything to stop those tools from loading them..but we've discussed eagerly warning if users have those config files.

Reply:
This is sensible. Shall I add a warning message if that file (`~/.config/yapf/style`) is found on a machine when `lint`/`fmt` goal is run? If so, could you please point me to a place where you already do that so that I could adopt it?